<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>900</width>
    <height>450</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>900</width>
    <height>450</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>900</width>
    <height>450</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Optimization</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QWidget" name="gridLayoutWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>401</width>
      <height>391</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="LayoutMain"/>
   </widget>
   <widget class="QGroupBox" name="BoxRS">
    <property name="geometry">
     <rect>
      <x>430</x>
      <y>10</y>
      <width>451</width>
      <height>101</height>
     </rect>
    </property>
    <property name="title">
     <string>Random Search properties</string>
    </property>
    <widget class="QRadioButton" name="radioButtonCount">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>20</y>
       <width>141</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Counter</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QRadioButton" name="radioButtonFail">
     <property name="geometry">
      <rect>
       <x>180</x>
       <y>20</y>
       <width>121</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Fail counter</string>
     </property>
    </widget>
    <widget class="QRadioButton" name="radioButtonDist">
     <property name="geometry">
      <rect>
       <x>340</x>
       <y>20</y>
       <width>101</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Distance</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="MaxCounter">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>70</y>
       <width>51</width>
       <height>20</height>
      </rect>
     </property>
     <property name="whatsThis">
      <string>Максимальное значение счетчика или числа неудач</string>
     </property>
     <property name="inputMethodHints">
      <set>Qt::ImhNone</set>
     </property>
     <property name="text">
      <string>1001</string>
     </property>
     <property name="maxLength">
      <number>32765</number>
     </property>
     <property name="placeholderText">
      <string>1000</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_3">
     <property name="geometry">
      <rect>
       <x>280</x>
       <y>40</y>
       <width>171</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Probability of local search</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_6">
     <property name="geometry">
      <rect>
       <x>190</x>
       <y>40</y>
       <width>81</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Distance</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_7">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>40</y>
       <width>121</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Counter</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="MinDist">
     <property name="geometry">
      <rect>
       <x>190</x>
       <y>70</y>
       <width>81</width>
       <height>22</height>
      </rect>
     </property>
     <property name="decimals">
      <number>5</number>
     </property>
     <property name="maximum">
      <double>1.000000000000000</double>
     </property>
     <property name="singleStep">
      <double>0.000001000000000</double>
     </property>
     <property name="value">
      <double>0.000010000000000</double>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="RSprob">
     <property name="geometry">
      <rect>
       <x>340</x>
       <y>70</y>
       <width>62</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maximum">
      <double>1.000000000000000</double>
     </property>
     <property name="singleStep">
      <double>0.010000000000000</double>
     </property>
     <property name="value">
      <double>0.500000000000000</double>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="BoxNM">
    <property name="geometry">
     <rect>
      <x>430</x>
      <y>120</y>
      <width>451</width>
      <height>81</height>
     </rect>
    </property>
    <property name="title">
     <string>Nelder Mead properties</string>
    </property>
    <widget class="QLabel" name="label_8">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>20</y>
       <width>61</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>alpha=</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_9">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>50</y>
       <width>51</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string> beta=</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_10">
     <property name="geometry">
      <rect>
       <x>310</x>
       <y>20</y>
       <width>61</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>gamma=</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_11">
     <property name="geometry">
      <rect>
       <x>310</x>
       <y>50</y>
       <width>61</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>  sigma=</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_12">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>20</y>
       <width>71</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Distance</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="NMdist">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>50</y>
       <width>91</width>
       <height>22</height>
      </rect>
     </property>
     <property name="decimals">
      <number>5</number>
     </property>
     <property name="maximum">
      <double>1.000000000000000</double>
     </property>
     <property name="value">
      <double>0.000010000000000</double>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="NMalpha">
     <property name="geometry">
      <rect>
       <x>210</x>
       <y>20</y>
       <width>62</width>
       <height>22</height>
      </rect>
     </property>
     <property name="whatsThis">
      <string>Коэффициент отражения(наихудшей вершины относительно центроида)</string>
     </property>
     <property name="minimum">
      <double>1.000000000000000</double>
     </property>
     <property name="maximum">
      <double>5.000000000000000</double>
     </property>
     <property name="value">
      <double>1.300000000000000</double>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="NMbeta">
     <property name="geometry">
      <rect>
       <x>210</x>
       <y>50</y>
       <width>62</width>
       <height>22</height>
      </rect>
     </property>
     <property name="whatsThis">
      <string>Коэффициент сжатия</string>
     </property>
     <property name="maximum">
      <double>1.000000000000000</double>
     </property>
     <property name="singleStep">
      <double>0.010000000000000</double>
     </property>
     <property name="value">
      <double>0.350000000000000</double>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="NMgamma">
     <property name="geometry">
      <rect>
       <x>370</x>
       <y>20</y>
       <width>62</width>
       <height>22</height>
      </rect>
     </property>
     <property name="whatsThis">
      <string>Коэффициент растяжения</string>
     </property>
     <property name="minimum">
      <double>1.000000000000000</double>
     </property>
     <property name="maximum">
      <double>5.000000000000000</double>
     </property>
     <property name="value">
      <double>2.500000000000000</double>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="NMsigma">
     <property name="geometry">
      <rect>
       <x>370</x>
       <y>50</y>
       <width>62</width>
       <height>22</height>
      </rect>
     </property>
     <property name="whatsThis">
      <string>Коэффициент сжатия вершин симплекса к найденному минимуму</string>
     </property>
     <property name="minimum">
      <double>0.010000000000000</double>
     </property>
     <property name="maximum">
      <double>1.000000000000000</double>
     </property>
     <property name="value">
      <double>0.750000000000000</double>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="BoxFunc">
    <property name="geometry">
     <rect>
      <x>430</x>
      <y>210</y>
      <width>451</width>
      <height>91</height>
     </rect>
    </property>
    <property name="title">
     <string>Function properties</string>
    </property>
    <widget class="QComboBox" name="comboBoxFunc">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>50</y>
       <width>271</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maxVisibleItems">
      <number>9</number>
     </property>
     <property name="maxCount">
      <number>3</number>
     </property>
     <item>
      <property name="text">
       <string>(x-2)^2+3*(x-y)^2</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>9*sin((x-y)^2)-cos((x+y)^2)+exp(y-x+1); </string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>y*exp(-x^2)+sin(x^2+y^2)</string>
      </property>
     </item>
    </widget>
    <widget class="QRadioButton" name="radioButtonRS">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>20</y>
       <width>171</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Random Search</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QRadioButton" name="radioButtonNM">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>20</y>
       <width>151</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Nelder Mead</string>
     </property>
    </widget>
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>50</y>
       <width>47</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>f(x,y)=</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>20</y>
       <width>61</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Method:</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>430</x>
      <y>310</y>
      <width>451</width>
      <height>91</height>
     </rect>
    </property>
    <property name="title">
     <string>Result</string>
    </property>
    <widget class="QLineEdit" name="ResultCounter">
     <property name="geometry">
      <rect>
       <x>180</x>
       <y>30</y>
       <width>51</width>
       <height>20</height>
      </rect>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLineEdit" name="Res_x">
     <property name="geometry">
      <rect>
       <x>270</x>
       <y>30</y>
       <width>61</width>
       <height>20</height>
      </rect>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLineEdit" name="Res_y">
     <property name="geometry">
      <rect>
       <x>370</x>
       <y>30</y>
       <width>61</width>
       <height>20</height>
      </rect>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLabel" name="label_4">
     <property name="geometry">
      <rect>
       <x>240</x>
       <y>30</y>
       <width>31</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>x=</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_5">
     <property name="geometry">
      <rect>
       <x>340</x>
       <y>30</y>
       <width>31</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>y=</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="Res_eval">
     <property name="geometry">
      <rect>
       <x>340</x>
       <y>60</y>
       <width>91</width>
       <height>20</height>
      </rect>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLabel" name="label_13">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>60</y>
       <width>47</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>f(x,y)=</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_14">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>30</y>
       <width>61</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>  Counter:</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="DomainFrom">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>60</y>
       <width>71</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>5.0; 5.0</string>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLineEdit" name="DomainTo">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>60</y>
       <width>71</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>-5.0; -5.0</string>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLabel" name="label_15">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>30</y>
       <width>101</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Current domain:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_16">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>60</y>
       <width>31</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>from</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_17">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>60</y>
       <width>31</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>to</string>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>900</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
